python/whylogs/api/writer/whylabs.py

```diff
@@ -557,8 +550,13 @@ def write(self, file: Writable, **kwargs: Any) -> Tuple[bool, str]:
             return self.write_feature_weights(file, **kwargs)
         elif isinstance(file, EstimationResult):
             return self.write_estimation_result(file, **kwargs)
-        elif isinstance(file, SegmentedResultSet) and self._reference_profile_name is not None:
-            return self._write_segmented_reference_result_set(file, **kwargs)
+        elif isinstance(file, ResultSet):
+            if isinstance(file, SegmentedResultSet) and self._reference_profile_name is not None:
+                return self._write_segmented_reference_result_set(file, **kwargs)
+                # TODO: why does segmented reference profile need special handling, but
+                # segmented non-reference profile and unsegmented reference profile just work?
+            file = file.profile()
```

Comment:
This throws on `SegmentedResultSet` :(

Reply:
The result sets have a get_writeables method, and this logic exists already in the result set writer.